galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt

```diff
@@ -656,43 +687,118 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
    * Returns the search conditions for this field.
    */
   open fun getSearchCondition(): String? {
-    return if (isNull(block!!.activeRecord)) {
-      when(getSearchOperator()) {
+    TODO()
+  }
+
+  /**
+   * Returns the search conditions for this field.
+   */
+  open fun getSearchCondition_(): Pair<Expression<String>.(t: String) -> Op<Boolean>, Any?>? {
+
+    if (isNull(block!!.activeRecord)) {
+      fun nullCondBuilder(body: (scope: Expression<String>) -> Op<Boolean>): (Expression<String>.(t: String) -> Op<Boolean>) {
+        return (fun Expression<String>.(t: String): Op<Boolean> = body(this))
+      }
+
+      return when (getSearchOperator()) {
         VConstants.SOP_EQ -> null
-        VConstants.SOP_NE -> "IS NOT NULL"
-        else -> "IS NULL"
+        VConstants.SOP_NE -> (nullCondBuilder {
+          IsNotNullOp(it)
+        } to null)
+        else -> (nullCondBuilder {
+          IsNullOp(it)
+        } to null)
       }
```

Comment:
here you return nullCondBuilder with type  (Expression<String>.(t: String) -> Op<Boolean>) but your method should return Pair<Expression<String>.(t: String) -> Op<Boolean>, Any?>? so check it !